pwndbg/dbg/lldb.py

```diff
@@ -9,22 +13,363 @@
+                    False,
+                    False,  # TODO: Handle base class members differently.
+                    field.bitfield_bit_size if field.is_bitfield else field.type.GetByteSize(),
+                )
+                for field in fields
+            ]
+            if len(fields) > 0
+            else None
+        )
+
+    @override
+    def array(self, count: int) -> pwndbg.dbg_mod.Type:
+        return LLDBType(self.inner.GetArrayType(count))
+
+    @override
+    def pointer(self) -> pwndbg.dbg_mod.Type:
+        return LLDBType(self.inner.GetPointerType())
+
+    @override
+    def strip_typedefs(self) -> pwndbg.dbg_mod.Type:
+        t = self.inner
+        while t.IsTypedefType():
+            t = t.GetTypedefedType
+
+        return LLDBType(t)
+
+    @override
+    def target(self) -> pwndbg.dbg_mod.Type:
+        t = self.inner.GetPointeeType()
+        if not t.IsValid():
+            raise pwndbg.dbg_mod.Error("tried to get target type of non-pointer type")
+
+        return LLDBType(t)
+
+
+class LLDBValue(pwndbg.dbg_mod.Value):
+    def __init__(self, inner: lldb.SBValue):
+        self.inner = inner
+
+    @property
+    @override
+    def address(self) -> pwndbg.dbg_mod.Value | None:
+        addr = self.inner.AddressOf()
+        return LLDBValue(addr) if addr.IsValid() else None
+
+    @property
+    @override
+    def is_optimized_out(self) -> bool:
+        return _is_optimized_out(self.inner)
+
+    @property
+    @override
+    def type(self) -> pwndbg.dbg_mod.Type:
+        assert not self.is_optimized_out, "tried to get type of optimized-out value"
+
+        return LLDBType(self.type)
+
+    @override
+    def dereference(self) -> pwndbg.dbg_mod.Value:
+        deref = self.inner.Dereference()
+
+        if not deref.IsValid():
+            raise pwndbg.dbg_mod.Error("could not dereference value")
+
+        return LLDBValue(deref)
+
+    @override
+    def string(self) -> str:
+        addr = self.inner.unsigned
+        error = lldb.SBError()
+
+        # Read strings up to 4GB.
+        last_str = None
+        buf = 256
+        for i in range(8, 33):
+            s = self.inner.process.ReadCStringFromMemory(addr, buf, error)
+            if error.Fail():
+                raise pwndbg.dbg_mod.Error(f"could not read value as string: {error.description}")
+            if last_str is not None and len(s) == len(last_str):
+                break
+            last_str = s
+
+            buf *= 2
+
+        return last_str
+
+    @override
+    def fetch_lazy(self) -> None:
+        # Not needed under LLDB.
+        pass
+
+    @override
+    def __int__(self) -> int:
+        return self.inner.signed
+
+    @override
+    def cast(self, type: pwndbg.dbg_mod.Type | Any) -> pwndbg.dbg_mod.Value:
+        assert isinstance(type, LLDBType)
+        t: LLDBType = type
+
+        return LLDBValue(self.inner.cast(t.inner))
+
+        return LLDBValue(self.inner.Cast(type.inner))
```

Comment:
Something is wrong here, the second `return` is dead code.

Reply:
Oh, yeah, right, that's probably a mistake of mine from when I did the rebase. Thanks for catching that!